CMakeLists.txt

```diff
@@ -318,6 +318,11 @@ if(NOT CORE_SYSTEM_NAME STREQUAL android)
   set_target_properties(${APP_NAME_LC} PROPERTIES ENABLE_EXPORTS ON)
 else()
   set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--no-undefined")
+  if(CPU MATCHES x86_64)
+    # Statically resolve global references to shared library (ie. ffmpeg) definitions.
+    # Related to https://stackoverflow.com/questions/46307266/including-objects-to-a-shared-library-from-a-c-archive-a
+    set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,-Bsymbolic")
```

Comment:
I'm not sure if this is the correct place. Wouldn't that apply everywhere?

Reply:
I don't like it there either, unfortunately Kodi is created on Android as shared libkodi.so and must unfortunately be used on it.

This annoys me too, that ffmpeg does not fix this. There have already been various users inquiries about where ffmpeg was referred to use this flag.

Reply:
Found to use this `set_target_properties(${APP_NAME_LC} PROPERTIES LINK_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,-Bsymbolic")` to make sure that really only on libkodi.so and nowhere else on his CMakeLists.txt becomes used.

Reply:
I thought of adding it as default linker flags everywhere, not only for the Kodi project or the kodi binary. Or could this cause different (linking) behaviour?

Reply:
During test it was mandatory to have on Kodi only, if this was e.g. on ffmpeg itself, becomes it to bad interact between both (compile works normal) and Kodi is not executable-